app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java

```diff
@@ -0,0 +1,128 @@
+package com.example.lhi06.residue;
+
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+
+public class AlbumAdapter extends RecyclerView.Adapter<AlbumAdapter.AlbumViewHolder> {
+
+    private ArrayList<Album> albumList = new ArrayList<>();
```

Comment:
The benefit being that List is the generic type and can be used to represent different types of lists, from ArrayList to ImmutableList etc. This way you can pass it into functions that modify the list without having to know which specific type it is. If you need me to clarify just ask